### PR TITLE
[Magiclysm] Overhaul and balance dragon items

### PR DESCRIPTION
**Pull request by Salty-Panda**
#### Summary
Mods "[Magiclysm] Overhaul and balance dragon items"

#### Purpose of change

It was in need of some changes, look below

#### Describe the solution

Materials:
- Reduce density, armor values, chip resistance from hide
- Add scale as material separate from the hide ( it was two items but sort of the same material ) with higher density, armor values, chip resistance than the hide.
- Hide armor is made of the hide only
- Scale armor is actually made of scales too now
- Make dragon hide and scales harder to burn down and immune to small (the lowest intensity ) fires

Recipes:
- Add activity level to recipes
- Add proficiencies for working with dragon leather and scales
- Add item quality `MAGIC_CAULDRON`
- Add another, faster, more efficient, and more magical recipe for `dragon essence`
- Add backpack ( and XL version ) made from dragon hide ( comparable to survivor backpack, but harder to destroy )

Balance:

- Reduce encumbrance, thickness from some hide armor, increase them for scale armor
- Increase warmth for scale armor
- Fix some values for XL items copy-pasted from normal ones ( encumbrance and crafting times )
- Remake scale helmet ( XL one too ) to have visor you can raise ( same like riot helmet ), reduce its eye and mouth encumbrance, add anti-glare quality ( like snow googles )
- Make crafting times in line with other hide/scale dragon armors and based on body parts ( e.g. gloves take longer to make )

#### Describe alternatives you've considered

Just balancing hide and scale armor against each other as I planned.

#### Testing

Loaded game and spawned/examined items and recipes
I expect Korg to give opinions about values and balance

#### Additional context

The backpack and armors both have no specified armor values and inherit them from the material.
But the inheritance in the case of backpack fails for acid and fire protection.

~~It fails build. I believe it's a separate error not caused by me: #47513.~~
Made visior of XL helmet have the same thickness as in normal helmet.

### CODE REVIEW

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 1]
```diff
@@ -144,7 +159,7 @@
     "name": "black dragonhide helmet",
     "description": "A helmet made from black dragonhide.  It protects your head well, and doesn't cover your face.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/black_dragon_items.json:160]:
```suggestion
    "description": "A helmet made from black dragonhide.  It protects your head well, but doesn't cover your face.",
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 2]
```diff
@@ -290,59 +305,31 @@
...
   {
-    "id": "helmet_xlblack_dragon_scale",
-    "copy-from": "helmet_black_dragon_scale",
+    "id": "helmet_xlblack_dragon_scale_raised",
+    "copy-from": "helmet_black_dragon_scale_raised",
     "type": "ARMOR",
-    "name": { "str": "XL black dragonscale helmet" },
-    "description": "A massive helmet made from black dragonscale, held together with black dragonhide.  It comes equipped with a full face visor and is large enough to fit even the strangest of heads.",
+    "name": { "str": "XL black dragonscale helmet (raised visor)", "str_pl": "XL black dragonscale helmets (raised visor)" },
+    "description": "A massive helmet made from black dragonscale, held together with black dragonhide.  It is large enough to fit even the strangest of heads. The visor is raised",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/black_dragon_items.json:328]:
```suggestion
    "description": "A massive helmet made from black dragonscale, held together with black dragonhide.  It is large enough to fit even the strangest of heads.  The visor is raised",
```

**Review by actual-nh — COMMENTED** [review]
A couple of things; I haven't looked over it for balance.

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 3]
```diff
@@ -119,32 +119,47 @@
...
   {
     "id": "helmet_black_dragon_hide",
     "type": "ARMOR",
     "category": "armor",
     "name": "black dragonhide helmet",
-    "description": "A helmet made from black dragonhide.  It protects your head well, and doesn't cover your face.",
+    "description": "A helmet made from very durable black dragonhide.  It protects your head well, but doesn't cover your face.",
     "weight": "585 g",
-    "volume": "2500 ml",
+    "volume": "2000 ml",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/black_dragon_items.json:162]:
```suggestion
    "volume": "2 L",
```

**Pull request merged**